app/scripts/modules/core/src/domain/IApplication.ts

```diff
@@ -0,0 +1,22 @@
+import { IIndexed } from './IIndexed';
+
+export interface IApplication extends IIndexed {
```

Comment:
I feel like we're conflating search results with actual domain items here (same with cluster, instance, etc.).

Can we rename these if they're new, or create separate classes to be more clear what they are instead of combining them with the existing domain classes?